packages/block-library/src/navigation-overlay-close/edit.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * WordPress dependencies
+ */
+import { Button, Icon, ToggleControl, PanelBody } from '@wordpress/components';
+import { close } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+import { useBlockProps, InspectorControls } from '@wordpress/block-editor';
+
+export default function Edit( { attributes, setAttributes } ) {
+	const blockProps = useBlockProps();
+	const { hasIcon } = attributes;
+
+	return (
+		<>
+			<InspectorControls>
+				<PanelBody title={ __( 'Display' ) }>
+					<ToggleControl
+						__nextHasNoMarginBottom
+						label={ __( 'Use icon' ) }
+						help={ __(
+							'Configure whether the button use an icon or text.'
+						) }
+						onChange={ ( value ) =>
+							setAttributes( { hasIcon: value } )
+						}
+						checked={ hasIcon }
+					/>
+				</PanelBody>
+			</InspectorControls>
+			<Button
+				{ ...blockProps }
+				aria-label={ hasIcon && __( 'Close menu' ) }
+			>
+				{ hasIcon ? <Icon icon={ close } /> : __( 'Close' ) }
+			</Button>
```

Comment:
I don't think we should use the `Button` component for actual block markup. I think this is only meant for admin UI. We should use the browser primitive itself. 

```suggestion
			<button
				{ ...blockProps }
				aria-label={ hasIcon && __( 'Close menu' ) }
			>
				{ hasIcon ? <Icon icon={ close } /> : __( 'Close' ) }
			</button>
```

Reply:
It's been a while...

Mind you even with this change I can't seem to get it to render on the front of my site 😓